### PR TITLE
[WIP] screen.c cleaning

### PR DESCRIPTION
**Pull request by romgrk**
- Cleanup variables in `win_line()`
 - Refactor `win_line()`

### CODE REVIEW

File: src/nvim/screen.c [hunk 1]
```diff
@@ -5389,66 +5247,41 @@ void screen_puts_len(char_u *text, int textlen, int row, int col, int attr)
...
+          ScreenLinesC[i][off] = u8cc[i];
+          if (u8cc[i] == 0)
+            break;
         }
-        screen_char(off, row, col);
-      } else if (mbyte_cells == 2) {
-        ScreenLines[off + 1] = ptr[1];
+      }
+      if (mbyte_cells == 2) {
+        ScreenLines[off + 1] = 0;
```
Comment (teto) [inline src/nvim/screen.c]: why not ptr[1] anymore ?

**Review by teto — APPROVED** [review]
Anything that makes screen.c more readable is gold.
You might want to add braces according to neovim style to please the linter.

File: src/nvim/screen.c (hunk 1 above)
Comment (romgrk) [inline src/nvim/screen.c]: What is left here is the inner body of the `if (l_enc_utf8)` starting at line 5327 (left). So the `if (mbyte_cells == 2)` here is the one at line 5340 (left), not the replacement of `else if (mbyte_cells == 2)` by `if (mbyte_cells == 2)` as suggested by the highlight.

File: src/nvim/screen.c [hunk 2]
```diff
@@ -2081,6 +2058,151 @@ fill_foldcolumn (
   }
 }
 
+
+static int get_column_diff_until_eol(win_T* wp, int col) {
```
Comment (romgrk) [inline src/nvim/screen.c]: I don't have a clear opinion here, I'll leave it as is unless someone strongly opposes to that name.

**Comment by justinmk** [comment]
> Removes references to enc_utf8, enc_dbcs & has_mbyte

How about doing that first, then leave the other changes for a separate PR.

**Comment by romgrk** [comment]
Sure, I've picked those commit and created a separate PR, I'll base this one on that one: https://github.com/neovim/neovim/pull/6609

**Comment by justinmk** [comment]
This should be done in smaller steps, starting from current HEAD. Also it's not necessarily a good idea to create tons of small functions, but factoring out a couple medium-sized ones would be worthwhile.

**Pull request closed without merging**